packages/service-library/tests/test_utils_secrets.py

```diff
@@ -50,3 +54,14 @@ def test_generate_passcode():
 
     # passcode is a number
     assert int(generate_passcode()) >= 0
+
+
+def test_compare_secrets():
+    passcode = generate_passcode(100)
+    assert not compare_secrets(got="foo", expected=passcode)
+    assert compare_secrets(got=passcode, expected=passcode)
```

Comment:
of passcode?
I rather not. This way I make sure `compare_secrets` do not change any of its inputs. 
In any case, it is also a string which is unmutable so it would not make a differece, right? :-)